### ISSUE TITLE
ArduinoOSC requires Ethernet library even when only WiFi is used

### ISSUE THREAD

**Issue by superkartoffel** [hideakitai/ArduinoOSC#19]
Stumbled upon this when I tried your library for the first time. Not sure if you consider this a bug, but since Ethernet is not stated as a required dependency, I do :)

steps to reproduce in Arduino IDE:
* make sure you have Ethernet lib NOT installed
* load arduino/OscWiFi example project
* select any board that supports Ethernet (e.g. ESP32)
* try to compile

Expected result:
It should compile without errors.

Actual result:
```
In file included from /home/kartoffel/Arduino/libraries/ArduinoOSC/examples/arduino/OscWiFi/OscWiFi.ino:2:0:
/home/kartoffel/Arduino/libraries/ArduinoOSC/ArduinoOSC.h:54:22: fatal error: Ethernet.h: No such file or directory
compilation terminated.
```

**Comment by superkartoffel** [comment]
A possible solution could be to split the header up into
ArduinoOSC_WiFi.h
ArduinoOSC_Ethernet.h

which set some defines and then include
ArduinoOSC.h

**Comment by hana** [comment]
Just sharing, this issue also happened in my environment, where PlatformIO, esp-idf, and Arduino (as a component) is used. 
So I tried installing the latest(v2.0.0) [Ethernet Library](https://github.com/arduino-libraries/Ethernet), ending up with a compile error on it. 

Since this environment might not be supported, I agree that it wouldn't work straightforward but I wish it would be supported!

**Comment by hideakitai** [comment]
@superkartoffel @hana Thank you for reporting, fixed in v0.3.22!
